tests/store/tracking/test_sqlalchemy_store.py

```diff
@@ -4167,25 +4170,55 @@ def _create_trace(
     if not store.get_experiment(experiment_id):
         store.create_experiment(store, experiment_id)
 
-    with mock.patch(
-        "mlflow.store.tracking.sqlalchemy_store.generate_request_id",
-        side_effect=lambda: request_id,
-    ):
-        trace_info = store.start_trace(
+    with store.ManagedSessionMaker() as session:
+        experiment = store.get_experiment(experiment_id)
+        if experiment.lifecycle_stage != LifecycleStage.ACTIVE:
+            raise MlflowException(
+                (
+                    f"The experiment {experiment.experiment_id} must be in the 'active' state. "
+                    f"Current state is {experiment.lifecycle_stage}."
+                ),
+                INVALID_PARAMETER_VALUE,
+            )
+
+        trace_info = SqlTraceInfo(
+            request_id=request_id,
             experiment_id=experiment_id,
             timestamp_ms=timestamp_ms,
-            request_metadata=request_metadata or {},
-            tags=tags or {},
+            execution_time_ms=None,
+            status=TraceStatus.IN_PROGRESS,
         )
 
-    store.end_trace(
-        request_id=request_id,
-        timestamp_ms=timestamp_ms + execution_time_ms,
-        status=status,
-        request_metadata={},
-        tags={},
-    )
-    return trace_info
+        trace_info.tags = [SqlTraceTag(key=k, value=v) for k, v in (tags.items() if tags else {})]
+        # Trace data is stored as file artifacts regardless of the tracking backend choice.
+        # We use subdirectory "/traces" under the experiment's artifact location to isolate
+        # them from run artifacts.
+        artifact_uri = append_to_uri_path(
+            experiment.artifact_location,
+            SqlAlchemyStore.TRACE_FOLDER_NAME,
+            request_id,
+            SqlAlchemyStore.ARTIFACTS_FOLDER_NAME,
+        )
+        artifact_location_tag = SqlTraceTag(key=MLFLOW_ARTIFACT_LOCATION, value=artifact_uri)
+
+        trace_info.tags.append(artifact_location_tag)
+        trace_info.request_metadata = [
+            SqlTraceRequestMetadata(key=k, value=v)
+            for k, v in (request_metadata.items() if request_metadata else {})
+        ]
```

Comment:
How do we ensure this is consistent with the actual SQL store implementation?

Reply:
I'd vote for Option 2. As this would be an internal-only configuration, this should not be exposed as a named argument, nor should this modify the public API interface.